index.html

```diff
@@ -5185,6 +5185,142 @@ <h2>Encrypting</h2>
     </section>
   </section>
 
+  <section class="appendix">
+    <h2>Vocabulary Overview</h2>
+
+    <p>This section provides an overview of all core properties defined in this specification in Section <a href="#core-properties"></a>, including the constraints on the usage of the properties.</p>
+
+    <section>
+      <h2>Classes</h2>
+
+      <p>The classes used in this section are not used DID Documents, and are only used to provide a more concise overview of the relevant core properties below. The are all represented, in a DID Document, as a <a data-cite="INFRA#maps">map</a> with specific properties.</p>
+
+      <table class=simple>
+        <thead>
+          <tr>
+            <th>Class</th>
+            <th>Description</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <td><code>DIDDocument</code></td>
+            <td>Class representing the “top level” of the <a>DID Document</a>. Each instance MUST have an <code>id</code> property, and MAY have the <code>alsoKnownAs</code>, <code>controller</code>, <code>verificationMethod</code>, <code>authentication</code>, <code>assertionMethod</code>, <code>keyAgreement</code>, <code>capabilityInvocation</code>, <code>capabilityDelegation</code>, and <code>service</code> properties.</td>
+          </tr>
+          <tr>
+            <td><code>VerificationMethod</code></td>
+            <td>Class representing the generic concept of <a href="#verification-methods"></a>. Each instance MUST have the <code>id</code>, <code>type</code>, and <code>controller</code> properties, as well as <em>one of</em> the <code>publicKeyJwk</code> or <code>publicKeyBase58</code> properties.</td>
+          </tr>
+          <tr>
+            <td><code>ServiceEndpoint</code></td>
+            <td>Class representing the generic concept of <a href="#service-endpoints"></a>. Each instance MUST have the <code>id</code>, <code>type</code>, and <code>serviceEndpoint</code> properties.</td>
+          </tr>
+      </table>
+    </section>
+
+    <section>
+      <h2>Properties</h2>
+
+      <p>Properties in the section are formally defined in Section <a href="#core-properties"></a>.</p>
+
+      <table class=simple>
+        <thead>
+          <tr>
+            <th>Property</th>
+            <th>Usage constraints</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <td><code>id</code></td>
+            <td>MUST be present for any <code>DIDDocument</code>, <code>VerificationMethod</code>, or <code>ServiceEndpoint</code> instance. The value MUST be
+              <ul>
+                <li>for a <code>DIDDocument</code>: a <a data-cite="INFRA#string">string</a> that conforms to the rules in Section <a href="#did-syntax"></a>;</li>
```

Comment:
I'm concerned that introducing the idea of classes in general but this class in particular might lead to more confusion. I feel like the next logical step is people saying "oh, well in that case `@type: DIDDocument` etc..

Reply:
Yeah, and that is also why I added an extra note in the text. But, unfortunately, if we do not use something like that then the language would become extremely convoluted (how would I describe the constraints on the `id` property, for example?) Any idea?